aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -280,11 +300,12 @@ private void completeMultiPartUpload() {
         .run(s3::completeMultipartUpload);
   }
 
-  private void abortUpload() {
+  private synchronized void abortUpload() {
```

Comment:
Thanks for noticing the multipartUploadId is not set back to null.

But I don't think we need to make it synchronized. If two threads both try to abort, one will just fail, and the finally block also does not throw exception. Do you see any issue when it is not synchronized?

Reply:
After reviewing the `abortMultipartUpload` API javadoc. We actually shouldn't set `multipartUploadId` back to `null`. Then we also don't need the `synchronized`

Reply:
Cool, I did not notice that. Could you also add a comment for explanation in the javadoc?
 > If any part uploads are currently in progress, those part uploads might or might not succeed. As a result, it might be necessary to abort a given multipart upload multiple times in order to completely free all storage consumed by all parts.

Reply:
already added the comment in the javadoc